### ISSUE TITLE
Crash on Linux: segfault 

### ISSUE THREAD

**Issue by houtaru** [CloudCompare/CloudCompare#1390]
**How to reproduce**
I open it and receive errors as below:

**Additional context**
Result of `journalctl`
```
Jan 28 00:27:44 houtaru kernel: CloudCompare[34835]: segfault at 0 ip 00007f4db3e26a11 sp 00007ffd641605b0 error 4 in libQt5XcbQpa.so.5.9.5[7f4db3d86000+fe000]
Jan 28 00:27:44 houtaru kernel: Code: e8 d4 2c f9 ff 48 85 db 75 e7 4c 89 ea 4c 89 f6 31 ff e8 f2 2d f9 ff 4c 89 ef 49 89 c7 e8 d7 20 f9 ff 4c 89 f7 e8 9f f0 f8 ff <41> 8b 07 85 c0 7e 21 45 >
Jan 28 00:27:44 houtaru kernel: audit: type=1701 audit(1611768464.286:517): auid=1000 uid=1000 gid=1000 ses=2 subj==snap.cloudcompare.CloudCompare (enforce) pid=34835 comm="CloudCompare" exe>
Jan 28 00:27:44 houtaru kernel: audit: type=1334 audit(1611768464.296:518): prog-id=113 op=LOAD
Jan 28 00:27:44 houtaru kernel: audit: type=1334 audit(1611768464.296:519): prog-id=114 op=LOAD
Jan 28 00:27:44 houtaru kernel: audit: type=1130 audit(1611768464.299:520): pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-coredump@11-34941-0 comm="systemd" e>
Jan 28 00:27:44 houtaru audit[34835]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=2 subj==snap.cloudcompare.CloudCompare (enforce) pid=34835 comm="CloudCompare" exe="/snap/cloudcompare/208/bi>
Jan 28 00:27:44 houtaru audit: BPF prog-id=113 op=LOAD
Jan 28 00:27:44 houtaru audit: BPF prog-id=114 op=LOAD
Jan 28 00:27:44 houtaru audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-coredump@11-34941-0 comm="systemd" exe="/usr/lib/systemd/systemd>
Jan 28 00:27:44 houtaru systemd[1]: Started Process Core Dump (PID 34941/UID 0).
Jan 28 00:27:47 houtaru systemd-coredump[34942]: Process 34835 (CloudCompare) of user 1000 dumped core.

                                                 Stack trace of thread 34835:
                                                 #0  0x00007f4db3e26a11 n/a (libQt5XcbQpa.so.5 + 0xa0a11)
                                                 #1  0x00007f4dca4b2b31 n/a (libQt5Gui.so.5 + 0x1d4b31)
                                                 #2  0x00007f4dca4b61ad _ZN13QFontDatabase8findFontERK8QFontDefi (libQt5Gui.so.5 + 0x1d81ad)
                                                 #3  0x00007f4dca4b69d6 _ZN13QFontDatabase4loadEPK12QFontPrivatei (libQt5Gui.so.5 + 0x1d89d6)
                                                 #4  0x00007f4dca48c433 _ZNK12QFontPrivate15engineForScriptEi (libQt5Gui.so.5 + 0x1ae433)
                                                 #5  0x00007f4dca4ac533 _ZNK12QFontMetrics6heightEv (libQt5Gui.so.5 + 0x1ce533)
                                                 #6  0x00007f4dcad46142 _ZN10QStatusBar8reformatEv (libQt5Widgets.so.5 + 0x2ff142)
                                                 #7  0x00007f4dcad463ce _ZN10QStatusBar18setSizeGripEnabledEb (libQt5Widgets.so.5 + 0x2ff3ce)
                                                 #8  0x0000563f3e4e9b16 _ZN13Ui_MainWindow7setupUiEP11QMainWindow (CloudCompare + 0x2e9b16)
                                                 #9  0x0000563f3e4c770c _ZN10MainWindowC1Ev (CloudCompare + 0x2c770c)
                                                 #10 0x0000563f3e4c8105 _ZN10MainWindow11TheInstanceEv (CloudCompare + 0x2c8105)
                                                 #11 0x0000563f3e33fdcc main (CloudCompare + 0x13fdcc)
                                                 #12 0x00007f4dc8c55bf7 n/a (/lib/x86_64-linux-gnu/libc-2.27.so + 0x21bf7)
```

**Your environment**
 - CC Version (or git hash/tag): 2.11.1 stable (install by snap)
 - OS & Version: Manjaro KDE 5.19.5
 - Graphic card: Intel Corporation UHD Graphics 620

**Comment by LeonDBL** [comment]
To add to Houtaru's problem, I can confirm the same on both the snap and AUR versions of CC. Flatpak "seems" to be okay so far but I'm yet to test fully.

I had segfaults from the flatpak and AUR on EndeavourOS.

Pop!_OS 20.10 was running the flatpak fine (just occasional crashes under load).

` Mar 14 13:00:14 manjaro kernel: CloudCompare[96449]: segfault at 200064 ip 00007f8dd4e57365 sp 00007ffd6efca290 error 4 in libqgnomeplatform.so[7f8dd4e40000+2e000]
Mar 14 13:00:14 manjaro kernel: Code: 89 ef ff 15 55 ab 02 00 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 00 41 54 49 89 fc 55 48 83 ec 08 48 85 f6 74 31 48 8b 16 48 89 17 <8b> 02 83 c0 01 83 f8 01 77 41 48 83 c6 08 49 8d 7c 24 08 ff 15 f2
Mar 14 13:00:14 manjaro kernel: audit: type=1701 audit(1615726814.440:657): auid=1000 uid=1000 gid=1000 ses=10 subj==unconfined pid=96449 comm="CloudCompare" exe="/usr/bin/CloudCompare" sig=11 res=1
Mar 14 13:00:14 manjaro audit: BPF prog-id=107 op=LOAD
Mar 14 13:00:14 manjaro audit: BPF prog-id=108 op=LOAD
Mar 14 13:00:14 manjaro systemd[1]: Started Process Core Dump (PID 96515/UID 0).
Mar 14 13:00:14 manjaro audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-coredump@9-96515-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Mar 14 13:00:14 manjaro kernel: audit: type=1334 audit(1615726814.457:658): prog-id=107 op=LOAD
Mar 14 13:00:14 manjaro kernel: audit: type=1334 audit(1615726814.457:659): prog-id=108 op=LOAD
Mar 14 13:00:14 manjaro kernel: audit: type=1130 audit(1615726814.457:660): pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-coredump@9-96515-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Mar 14 13:00:15 manjaro systemd-coredump[96516]: Process 96449 (CloudCompare) of user 1000 dumped core.
                                                 
                                                 Stack trace of thread 96449:
                                                 #0  0x00007f8dd4e57365 _ZN17QtMetaTypePrivate23QMetaTypeFunctionHelperIN27QXdgDesktopPortalFileDialog6FilterELb1EE9ConstructEPvPKv (libqgnomeplatform.so + 0x2f365)
                                                 #1  0x00007f8dd91d2b90 n/a (libQt5Core.so.5 + 0x2f5b90)
                                                 #2  0x00007f8dd91d2eae _ZN8QVariantC2EiPKvj (libQt5Core.so.5 + 0x2f5eae)
                                                 #3  0x00007f8dd4e5514c _ZN27QXdgDesktopPortalFileDialog10openPortalEv (libqgnomeplatform.so + 0x2d14c)
                                                 #4  0x00007f8dd4e56058 _ZN27QXdgDesktopPortalFileDialog4showE6QFlagsIN2Qt10WindowTypeEENS1_14WindowModalityEP7QWindow (libqgnomeplatform.so + 0x2e058)
                                                 #5  0x00007f8dd9e5eeea _ZN14QDialogPrivate22setNativeDialogVisibleEb (libQt5Widgets.so.5 + 0x35ceea)
                                                 #6  0x00007f8dd9e72068 _ZN11QFileDialog10setVisibleEb (libQt5Widgets.so.5 + 0x370068)
                                                 #7  0x00007f8dd9e60117 _ZN7QDialog4execEv (libQt5Widgets.so.5 + 0x35e117)
                                                 #8  0x00007f8dd9e731f5 _ZN11QFileDialog15getOpenFileUrlsEP7QWidgetRK7QStringRK4QUrlS4_PS2_6QFlagsINS_6OptionEERK11QStringList (libQt5Widgets.so.5 + 0x3711f5)
                                                 #9  0x00007f8dd9e73312 _ZN11QFileDialog16getOpenFileNamesEP7QWidgetRK7QStringS4_S4_PS2_6QFlagsINS_6OptionEE (libQt5Widgets.so.5 + 0x371312)
                                                 #10 0x00005558f68b1c4e _ZN10MainWindow16doActionLoadFileEv (CloudCompare + 0x2dec4e)
                                                 #11 0x00007f8dd91c9d86 n/a (libQt5Core.so.5 + 0x2ecd86)
                                                 #12 0x00007f8dd9c55f63 _ZN7QAction9triggeredEb (libQt5Widgets.so.5 + 0x153f63)
                                                 #13 0x00007f8dd9c58845 _ZN7QAction8activateENS_11ActionEventE (libQt5Widgets.so.5 + 0x156845)
                                                 #14 0x00007f8dd9d50142 n/a (libQt5Widgets.so.5 + 0x24e142)
                                                 #15 0x00007f8dd9d50263 _ZN15QAbstractButton17mouseReleaseEventEP11QMouseEvent (libQt5Widgets.so.5 + 0x24e263)
                                                 #16 0x00007f8dd9e4501b _ZN11QToolButton17mouseReleaseEventEP11QMouseEvent (libQt5Widgets.so.5 + 0x34301b)
                                                 #17 0x00007f8dd9c9db0e _ZN7QWidget5eventEP6QEvent (libQt5Widgets.so.5 + 0x19bb0e)
                                                 #18 0x00007f8dd9c5c752 _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent (libQt5Widgets.so.5 + 0x15a752)
                                                 #19 0x00007f8dd9c6387b _ZN12QApplication6notifyEP7QObjectP6QEvent (libQt5Widgets.so.5 + 0x16187b)
                                                 #20 0x00007f8dd9192a2a _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent (libQt5Core.so.5 + 0x2b5a2a)
                                                 #21 0x00007f8dd9c6287e _ZN19QApplicationPrivate14sendMouseEventEP7QWidgetP11QMouseEventS1_S1_PS1_R8QPointerIS0_Ebb `

- CC: Version 2.11 (snap and AUR)
- OS: Manjaro Gnome
- GPU: AMD Vega 56

**Comment by luca-penasa** [comment]
Hi @LeonDBL , 
could you try out the cloudcompare-git package here https://github.com/luca-penasa/AUR-cloudcompare-git
Those AUR packages might need to be updated probably.

luca

**Comment by LeonDBL** [comment]
Hi @luca-penasa 

Apologies for my delay in reply.

OK, just installed again via AUR - no joy (on Manjaro Gnome). CloudCompare loads, but if I go to open a file I get;

`QSocketNotifier: Can only be used with threads started with QThread
"/proc/21495/root"
zsh: segmentation fault (core dumped)  CloudCompare`

**Comment by luca-penasa** [comment]
I am on Manjaro too, and for me, it is working fine. If I can do anything to check my environment or similar to see how well it compares to yours just let me know.

What I would suggest is to double-check that no "old" cloud compare libraries were forgotten around in your system. e.g. from previous compilations from sources. (this is something that happens to me and it leads to segfault all the time).

An easy way to so Is probably to try to lunch CC from a new and "clean" user (logout and login as a different user) or carefully check local variables (e.g. local definitions of LD_LIBRARY_PATH or similar in your .bashrc etc) - that would exclude some local folders but will not help if you previously did any "make install" into the system. 

Sorry, it is not much, and I am not sure it might help... let me know if I can do anything more.

of course, also compiling cc in debug mode and then run it through gdb would help...

luca

**Comment by dgirardeau** [comment]
I'm definitely not a Linux expert, but as it's crashing in the Qt5X / Qt5Gui modules, can it be related to the display driver?

**Comment by LeonDBL** [comment]
@luca-penasa Interesdting, OK, I have a test environment for Manjaro so I'll purge any CloudCompare install from that and test again as instructed. Out of curiosity, which DE are you using?

@dgirardeau I did wonder that myself, on my Sway DE install, CloudCompare installs and runs, but the display output looks like I have an error with my GPU (which I know there's not). Doesn't seem to like Wayland, or certainly not within that particular setup. I'm currently running it on X within Gnome, and I was thinking about testing to see if a different DE that uses X resolves any issue. Manjaro seems pretty well put together for rendering QT and GTK applications, but I've still had to install additional QT5 libraries for following Luca's build instructions above.

I'll do some testing this weekend.

**Comment by luca-penasa** [comment]
I am on gnome actually, but not on wayland. I know it might have problems here and there, I would try to login in gnome without it if possible.

**Comment by LeonDBL** [comment]
Test system (Sway and Gnome on Manjaro): Purged CloudCompare, updated system as updates available today (quite a few  including some X and qt stuff), logged into Gnome on X, installed CloudCompare 2.11 from the normal AUR - working perfectly. Logged into Sway (Wayland) working perfectly.

Daily system (Gnome on Manjaro): Purged CloudCompare, updated system (quite a few updates as above), logged into Gnome on X, installed CloudCompare 2.11 from the normal AUR - working, but pdal is failing to locate laszip to install las files, but that seems to be an install issue as it's working elsewhere.

TL:DR - it's working again on X and Wayland. ¯\_(ツ)_/¯ (happy days!)

**Comment by luca-penasa** [comment]
great!
